cli_tools/google-osconfig-agent/main.go

```diff
@@ -83,6 +102,17 @@ func main() {
 	flag.Parse()
 	ctx := context.Background()
 
+	// TODO: move to a different locking system or move lockFile definition to config.
```

Comment:
Consider moving this to a separate function that makes its purpose clearer, for example `ensureSingletonProcess` or `ensureOsConfigNotAlreadyRunning`.

Also, fwiw, this seems like a brittle way to do it since the process could crash w/out removing the lock file. We may need to put a timestamp in the file that the process with the lock continually updates...

Reply:
Oh. Does `O_EXCL` release/delete the file if the process terminates?

Reply:
Nope, O_EXCL just makes sure the file doesnt exist already. The point of my TODO was that we should find a better solution later, I'll update it to be more clear on the reasons.

Also i feel naming is totally fine, your examples are crazy verbose :) this is all private to main which has like 3 functions :p

Reply:
Does that mean that if the process crashes and doesn't release the lock, the agent won't be able to restart? If so, I don't think we'd want to use this mechanism even for a little while. Can we use `https://golang.org/pkg/syscall/#Flock` instead?

Reply:
flock only works on linux

Reply:
Implemented for windows and linux